nym-vpn-core/crates/nym-vpn-api-client/src/client.rs

```diff
@@ -82,24 +85,46 @@ impl VpnApiClient {
                 builder
             })
             .and_then(|builder| builder.build())
-            .map(|c| Self { inner: c })
+            .map(|c| Self { inner: c, vpn_api_time: Arc::new(RwLock::new(None)) })
             .map_err(VpnApiClientError::FailedToCreateVpnApiClient)
     }
 
+    pub fn swap_inner_client(&mut self, client: &VpnApiClient) {
+        self.inner = client.inner.clone();
+    }
+
     pub fn current_url(&self) -> &Url {
         self.inner.current_url()
     }
 
-    pub async fn get_remote_time(&self) -> Result<VpnApiTime> {
+    fn use_remote_time(remote_time: VpnApiTime) -> bool {
+        if remote_time.is_almost_same() {
+            tracing::debug!("{remote_time}");
+            false
+        } else if remote_time.is_acceptable_synced() {
+            tracing::info!("{remote_time}");
+            false
+        } else {
+            tracing::warn!(
+                "The time skew between the local and remote time is too large, we'll use remote instead for JWT ({remote_time})."
+            );
+            true
+        }
+    }
+
+    pub async fn sync_with_remote_time(&mut self) -> Result<()> {
         let time_before = OffsetDateTime::now_utc();
         let remote_timestamp = self.get_health().await?.timestamp_utc;
```

Comment:
nit: what about adding a `.inspect_err` here and then removing the `if let Err() = ... tracing::error` on the callsites? Since we always continue on error anyway

Reply:
That sounds good